Wangscape/TilePartitionerPerlin.cpp

```diff
@@ -0,0 +1,103 @@
+#include "TilePartitionerPerlin.h"
+#include "ModuleFactories.h"
+#include "NoiseMap.h"
+#include "AlphaCalculator.h"
+
+void TilePartitionerPerlin::makeCorner(NoiseMapVector<float>& noise_map_vector,
+                                       const Corners& corners,
+                                       bool left, bool top)
+{
+    TerrainID corner_id = corners[ (left ? 0 : 1) +  (top ? 0 : 2)];
+    TerrainID corner_h =  corners[(!left ? 0 : 1) +  (top ? 0 : 2)];
+    TerrainID corner_v =  corners[ (left ? 0 : 1) + (!top ? 0 : 2)];
+
+    Reseedable& stochastic_mask = mNoiseModuleManager.getStochastic(corner_id);
+    Reseedable border_h = mNoiseModuleManager.getBorderHorizontal(left ? corner_id : corner_h,
+                                                                  left ? corner_h : corner_id,
+                                                                  top);
+    Reseedable border_v = mNoiseModuleManager.getBorderVertical(top ? corner_id : corner_v,
+                                                                top ? corner_v : corner_id,
+                                                                left);
+    Reseedable cc = makeCornerCombiner(left,top);
+    Reseedable border_xy = cc.blend(border_v, border_h);
+    Reseedable deterministic = makeLinearMovingScaleBias(border_xy, left, top, 0.85, 0.15);
+    Reseedable ef = makeEdgeFavouringMask(1.5, 1.);
+    Reseedable corner = ef.blend(stochastic_mask, deterministic);
+#ifdef _DEBUG
+    sf::Image debug;
+    debug.create(mOptions.resolution, mOptions.resolution);
+    NoiseMapImage debug_map(debug, {0, 0, 1, 1});
+    auto write_map = [&](const Reseedable module,
+                         std::string filename,
+                         NoiseMapImage& nmi)
+    {
+        nmi.build(*module.module);
+        debug.saveToFile("debug/" + filename + ".png");
+    };
+
+    write_map(stochastic_mask, "stoch", debug_map);
+    write_map(border_h, "bh", debug_map);
+    write_map(border_v, "bv", debug_map);
+    write_map(cc, "cc", debug_map);
+    write_map(border_xy, "bhv", debug_map);
+    Reseedable max_det = makeLinearMovingScaleBias(makeConst(1.), left, top, 0.85, 0.15);
+    Reseedable min_det = makeLinearMovingScaleBias(makeConst(-1.), left, top, 0.85, 0.15);
+    write_map(min_det, "min_det", debug_map);
+    write_map(max_det, "max_det", debug_map);
+    write_map(deterministic, "det", debug_map);
+    write_map(ef, "ef", debug_map);
+    write_map(corner, "corner", debug_map);
+#endif
```

Comment:
Yes, my current task is to replace these lines with a more complete and flexible system for outputting debug tilesets. The user should be able to switch debug output on/off for each module, and debug output should be for full tilesets rather than single tiles.

In the meantime, these lines would be better in a function, and controlled by a command line flag rather than a compiler flag.